### PR TITLE
Make sqllogictest distinguish NaN from -NaN

### LINKED ISSUE #7402: sqllogictest should distinguish NaN from -NaN
### Describe the bug

The current sqllogictest doesn't distinguish `NaN` and `-NaN`.
As DataFusion distinguishes them, sqllogictest should distinguish them too.

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sarutak**
## Which issue does this PR close?
Closes #7402 

## Rationale for this change
DataFusion distinguishes `NaN` and `-NaN` so sqllogictest should distinguish them for precise test.

## What changes are included in this PR?
Changed `conversion.rs` so that it returns `"-NaN"` if a value is a kind of `NaN` and its sign is negative.

## Are these changes tested?
Modified existing test and added a simple test to `select.rs`.

## Are there any user-facing changes?
No.

### CODE REVIEW

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
pyarrow error looks unrelated. Maybe re-trigger it later.

File: datafusion/sqllogictest/test_files/math.slt [hunk 1]
```diff
@@ -95,10 +95,10 @@ SELECT atan2(2.0, 1.0), atan2(-2.0, 1.0), atan2(2.0, -1.0), atan2(-2.0, -1.0), a
 1.107148717794 -1.107148717794 2.034443935796 -2.034443935796 NULL NULL NULL
 
 # nanvl
-query RRR
-SELECT nanvl(asin(10), 1.0), nanvl(1.0, 2.0), nanvl(asin(10), asin(10))
+query RRB
+SELECT nanvl(asin(10), 1.0), nanvl(1.0, 2.0), isnan(nanvl(asin(10), asin(10)))
```
Comment (sarutak) [inline datafusion/sqllogictest/test_files/math.slt:99]: The sign of `NaN` returned by `asin(10)` seems different between operating systems.
So, I use `isnan` just to check if it's `NaN` or not.

**Comment by alamb** [comment]
I believe CI will pass after merging up from `main` to get the change in  https://github.com/apache/arrow-datafusion/pull/7416

**Comment by alamb** [comment]
I merged up from main to get the CI test fix

**Comment by berkaysynnada** [comment]
We have encountered a problem related to this PR. The sign bit of the value representing NaN is platform-dependent. (As seen in this [issue](https://github.com/apache/arrow-datafusion/issues/7458), I think the result of scalar functions on ARM is always NaN, not -NaN.). Can you suggest a way to solve this issue in a deterministic way? `isnan()` cannot solve the problem for columnar data.

**Comment by sarutak** [comment]
Hi @berkaysynnada 
Can we modify the query in sqllogictests using `nanvl`?

**Comment by sarutak** [comment]
@berkaysynnada  I opened #7462 for the issue you hit.
sqllogictest works on your machine with that change?

**Pull request merged**